calico/felix/endpoint.py

```diff
@@ -178,6 +291,49 @@ def on_interface_update(self, name, iface_up):
                 ep = self.objects_by_id[endpoint_id]
                 ep.on_interface_update(iface_up, async=True)
 
+    def _update_dirty_policy(self):
+        if not self._data_model_in_sync:
+            _log.debug("Datamodel not in sync, postponing update to policy")
+            return
+        _log.debug("Endpoints with dirty policy: %s",
+                   self.endpoints_with_dirty_policy)
+        while self.endpoints_with_dirty_policy:
+            ep_id = self.endpoints_with_dirty_policy.pop()
+            if self._is_starting_or_live(ep_id):
+                self._update_tiered_policy(ep_id)
+
+    def _update_tiered_policy(self, ep_id):
+        """
+        Sends an updated list of tiered policy to an endpoint.
+
+        Recalculates the list.
+        :param ep_id: ID of the endpoint to send an update to.
+        """
+        _log.debug("Updating policies for %s from %s", ep_id,
+                   self.pol_ids_by_ep_id)
+        # Order the profiles by tier and profile order, using the name of the
+        # tier and profile as a tie-breaker if the orders are the same.
+        profiles = []
+        for pol_id in self.pol_ids_by_ep_id.iter_values(ep_id):
+            try:
+                tier_order = self.tier_orders[pol_id.tier]
+                profile_order = self.profile_orders[pol_id]
+            except KeyError:
+                _log.warn("Ignoring profile %s because its tier metadata is "
+                          "missing.")
```

Comment:
This warning should indicate whether it's the tier order or the policy order that is missing (or both).  Also ideally improve the wording to cover both possibilities.